keps/sig-auth/4633-anonymous-auth-configurable-endpoints/README.md

```diff
@@ -0,0 +1,950 @@
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `AnonymousAuthConfigurableEndpoints`
+  - Components depending on the feature gate: `kube-apiserver`
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+Enabling the feature gate does not change the default behavior unless the user
+also changes the value of `--anonymous-auth` flag or updates the
+`AuthenticationConfiguration`.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+Yes.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing, unless the user also changes the value of `--anonymous-auth` flag or
+updates the `AuthenticationConfiguration`.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Yes we will add these tests.
```

Comment:
Can we enumerate some of the key things we'd like to test for?  I'm particularly interested in making sure we get this right for alpha so if we need to rollback from beta to alpha it is safe:

- When feature gate is disabled: `AuthenticationConfiguration` may contain Anonymous stanza, but it is ignored
- When feature gate is disabled:  `--anonymous-auth` may be set to `false`, but it is ignored
- <whatever other disabled/enablement tests you feel need to be included>

Reply:
Thanks! Updated. Please take another look.